### ISSUE TITLE
Fix subscripting

### ISSUE THREAD

**Issue by wiztigers** [TypeCobolTeam/TypeCobol#302]
**SPAWN OF #249**

The parser must be able to detect is an item is properly subscripted, as described by unit test `Programs/TypeCobol/Tables`.

**Comment by wiztigers** [comment]
Given the following declaration:
```cobol
01 T-fixed-1         OCCURS 100.
  05 T-fixed-2       OCCURS 100.
    10 E             PIC X(8).
```

What is the specific COBOL compiler reaction about the following statement ?
```cobol
MOVE X     TO E OF T-fixed-2 OF T-fixed-1 (1)
```
Does it complain about `T-fixed-1` not beging subscripted, or about `T-fixed-2` not being subscripted ?
From 5c8bc81, our parser scoffs about `T-fixed-2`, but it is just so I could use a "natural" `foreach` iteration in `DeclarationChecker.CheckSubscripting`. I can warn about `T-fixed-1` instread, by just used a reverse iteration on the same `LinkedList`.

**Comment by smedilol** [comment]
With this code:
```cobol
01  Tab.
   02 T-fixed-1         OCCURS 100.
     05 T-fixed-2       OCCURS 100.
        10 E             PIC X(8).
01 X pic X(08).
      
PROCEDURE DIVISION.
      MOVE X     TO E OF T-fixed-2 OF T-fixed-1 (1)
```
I get the following error:
**IGYPS2013-E** `Not enough subscripts or indices were specified for"E OF T-FIXED-2 OF T-FIXED-1".  A subscript or index value of 1 was assumed for each missing subscript or index.`

PS: You can't have an occurs on a level 01 variable.

**Comment by wiztigers** [comment]
Well, as this is an error and the parser doesn't have any default subscript or index value to assume, what we do here does suffice IMO.
In the context of a `OF` qualification, which specific item is not subscripted doesn't matter ; whereas in the context of a  `::` qualification, subscripting will work a tad differently that it works now, but it is in the sope of issue #307.
So, this issue can be closed.


Forbidden OCCURS clause on a level 01 item is, I suppose, among those 65535 strictly COBOL errors we haven't implemented, so I guess it's not even worth it to create an issue for it now ?
But @smedilol feel free to create such an issue if you don't agree, it'd be pretty easy to implement. Problem is just, if we begin like that, to track which COBOL errors we implemented and wich ones we did not.
